tests/xml/XMLBearTest.py

```diff
@@ -18,14 +20,59 @@
 <a>hey and hi</a>
 """.splitlines(keepends=True)
 
-InvalidXMLCorrectedTest = verify_local_bear(
-                                        XMLBear,
-                                        valid_files=(valid_xml_file,),
-                                        invalid_files=(invalid_xml_file,),
-                                        tempfile_kwargs={"suffix": ".xml"})
-
-InvalidXMLSyntaxTest = verify_local_bear(
-                                        XMLBear,
-                                        valid_files=(valid_xml_chars,),
-                                        invalid_files=(invalid_xml_chars,),
-                                        tempfile_kwargs={"suffix": ".xml"})
+dtd_file = os.path.join(os.path.dirname(__file__),
+                        "test_files",
+                        "note.dtd")
+
+schema_file = os.path.join(os.path.dirname(__file__),
+                           "test_files",
+                           "note.xsd")
+
+valid_xml_path = os.path.join(os.path.dirname(__file__),
+                              "test_files",
+                              "note.xml")
+
+valid_xml_url = os.path.join(os.path.dirname(__file__),
+                             "test_files",
+                             "concept-valid.xml")
+
+invalid_xml_schema = os.path.join(os.path.dirname(__file__),
+                                  "test_files",
+                                  "xsd-error.xml")
+
+invalid_xml_dtd = os.path.join(os.path.dirname(__file__),
+                               "test_files",
+                               "dtd-error.xml")
+
+invalid_xml_url = os.path.join(os.path.dirname(__file__),
+                               "test_files",
+                               "concept-invalid.xml")
+
+dtd_url = "http://docs.oasis-open.org/dita/v1.0.1/dtd/concept.dtd"
+
+XMLBearCorrectedTest = verify_local_bear(
+    XMLBear,
+    valid_files=(valid_xml_file, valid_xml_chars),
+    invalid_files=(invalid_xml_file, invalid_xml_chars),
+    tempfile_kwargs={"suffix": ".xml"})
+
+XMLBearSchemaTest = verify_local_bear(
+    XMLBear,
+    valid_files=(list(open(valid_xml_path, 'r')),),
+    invalid_files=(list(open(invalid_xml_schema, 'r')),),
+    settings={'xml_schema': schema_file},
+    tempfile_kwargs={"suffix": ".xml"})
+
+XMLBearDTDPathTest = verify_local_bear(
+    XMLBear,
+    valid_files=(list(open(valid_xml_path, 'r')),),
+    invalid_files=(list(open(invalid_xml_dtd, 'r')),),
+    settings={'xml_dtd': dtd_file},
```

Comment:
Only the dtd_file cannot be created as a variable because of https://github.com/coala-analyzer/coala-bears/issues/110